register-form/src/components/RegForm.vue

```diff
@@ -0,0 +1,186 @@
+
+<template>
+  <div>
+    <label>First name:</label><br>
+    <input :class="{invalid: !valids.firstName & !firstTime}" type="text" v-model="inputs.firstName" >
+    <br>
+    <span class="error" v-show="!valids.firstName & !firstTime">
+          *First name must have atleast one letter.</span>
+    <br>
+
+    <label>Last name:</label><br>
+    <input :class="{invalid: !valids.lastName & !firstTime}" type="text" v-model="inputs.lastName">
+        <br>
+    <span class="error" v-show="!valids.lastName & !firstTime">
+          *Last name must have atleast one letter.</span>
+    <br>
+
+    <label >Gender:</label> <br>
+    <select :class="{invalid: !valids.gender & !firstTime}" v-model="inputs.gender">
+      <option value="" selected>Gender</option>
+      <option value="male">Male</option>
```

Comment:
I like more the eNum approach - meaning that the presented string is the full one but the value we actually keep is 0/1/2/3 or ""/"M"/"F"/"B" - but that is a matter of taste and the effects of it on performance are neglligible, so again your choice